src/Http/Connection.php

```diff
@@ -39,17 +43,34 @@ public function __construct(?LoggerInterface $logger, array $clientOptions = [])
 
     public function get(string $url, RequestOptionsInterface $requestOptions, array $args = []) : ResponseInterface
     {
+        $this->requestArgs = [
+            'url' => $url,
+            'requestOptions' => $requestOptions,
+            'args' => $args,
+        ];
+
         $client = $this->createClientWithRetryHandler();
         $options = [
-            'headers' => $requestOptions->getHeaders()
+            'headers' => $requestOptions->getHeaders(),
+            'http_errors' => false,
         ];
 
         if (empty($args['skipQueryParams'])) {
             $options['query'] = $requestOptions->getQueryParams();
         }
 
         try {
-            return $client->request(RequestType::Get, $url, $options);
+            $response = $client->request(RequestType::Get, $url, $options);
+            if ($this->shouldRetry($response->getStatusCode())) {
```

Comment:
this method already uses a retry handler , may be we should extend `createRetryHandler ` that to include this as well?

Also shouldRetry has a lot of failure codes, can we check what status code we get when we don't have a valid token (prob 401, 403). In that case we don't have to retry again if the status code doesn't match an invalid token